rules/prefer-object-define-properties.js

```diff
@@ -0,0 +1,122 @@
+'use strict';
+const isSameReference = require('./utils/is-same-reference.js');
+const {removeArgument} = require('./fix/index.js');
+
+const MESSAGE_ID = 'prefer-object-define-properties';
+const messages = {
+	[MESSAGE_ID]: 'Prefer `{{replacement}}` over multiple `{{value}}`.',
+};
+
+function getFirstExpression(node, sourceCode) {
+	const {parent} = node;
+	const visitorKeys = sourceCode.visitorKeys[parent.type] || Object.keys(parent);
+
+	for (const property of visitorKeys) {
+		const value = parent[property];
+		if (Array.isArray(value)) {
+			const index = value.indexOf(node);
+
+			if (index !== -1) {
+				return value[index - 1];
+			}
+		}
+	}
+}
+
+/** @param {import('eslint').Rule.RuleContext} context */
+const create = context => ({
+	'Program:exit'() {
+		const sourceCode = context.getSourceCode();
+		const references = context
+			.getScope()
+			.variableScope.set.get('Object')
+			.references.filter(
+				reference =>
+					reference.identifier.parent.type === 'MemberExpression'
+					&& reference.identifier.parent.property.name === 'defineProperty'
+					&& reference.identifier.parent.parent.arguments?.[1]?.type === 'Literal'
+					&& reference.identifier.parent.parent.arguments?.[2]?.type === 'ObjectExpression',
+			)
+			.map(reference => ({
+				scope: reference.from.block,
+				node: reference.identifier.parent.parent,
+				arguments: reference.identifier.parent.parent.arguments,
+			}));
+
+		const groups = [];
+		let currentGroup;
+
+		for (const reference of references) {
+			const firstExpression = getFirstExpression(reference.node.parent, sourceCode);
+			const newGroup = firstExpression?.expression?.callee.object
```

Comment:
We should also check  `Object.defineProperties()` + `Object.defineProperty()`, not only `Object.defineProperty()` + `Object.defineProperty()`.

So we can also handle case that call `Object.defineProperty()` more than twice.